Corona-Warn-App/src/deviceForTesters/java/de/rki/coronawarnapp/test/submission/ui/SubmissionTestFragmentViewModel.kt

```diff
@@ -93,17 +93,19 @@ class SubmissionTestFragmentViewModel @AssistedInject constructor(
 
     fun submit() = launch {
         try {
-            srsSubmissionRepository.submit(
+            val result = srsSubmissionRepository.submit(
                 type = SrsSubmissionType.SRS_SELF_TEST,
                 keys = tekHistory.value.orEmpty().map { it.key }
             )
-            srsSubmissionResult.postValue(Success)
+
+            val event = when (result) {
+                SrsSubmissionResponse.Success -> Success
+                is SrsSubmissionResponse.TruncatedKeys -> TruncatedSubmission(result.days)
+            }
+            srsSubmissionResult.postValue(event)
         } catch (e: Exception) {
             Timber.e(e, "submit()")
-            when (e) {
-                is SrsSubmissionTruncatedException -> srsSubmissionResult.postValue(TruncatedSubmission(e.message))
```

Comment:
The class `SrsSubmissionTruncatedException` is still present but not in use anymore. I think you can remove that too.

Reply:
deleted